Content.Server/Mobs/Mind.cs

```diff
@@ -272,8 +275,15 @@ public void TransferTo(IEntity? entity)
             OwnedComponent = component;
             OwnedComponent?.InternalAssignMind(this);
 
-            if (VisitingEntity?.HasComponent<GhostComponent>() == false)
+            GhostComponent? ghostComponent = null;
+            VisitingEntity?.TryGetComponent(out ghostComponent);
+
+            if (ghostComponent == null || // visiting entity is not a Ghost
```

Comment:
You can probably replace some of these checks with the TryGetComponent<GhostComponent>()